mobile/src/Student/screens/PhaseResult/index.js

```diff
@@ -1,69 +1,62 @@
+      React.useCallback(() => {
+        gameSession?.currentState === GameSessionState.PHASE_1_RESULTS ? setPhaseNo(1) : setPhaseNo(2)  
 
-    useEffect(() => {
-        switch (gameSession.currentState) {
-            case GameSessionState.PHASE_1_RESULTS:
-                setPhaseNo(1)
-                break
-            case GameSessionState.PHASE_2_RESULTS:
-                setPhaseNo(2)
-                break
-        }
         const updatedCurTeam = gameSession.teams.find((t) => t.id === team.id)
         if (isNullOrUndefined(updatedCurTeam)) {
             console.error(`Couldn't find the team.}`)
             return
         }
         setCurTeam(updatedCurTeam)
-        const curQuestion = gameSession.questions[gameSession.currentQuestionIndex]
-        const teamAnswers = ModelHelper.getBasicTeamMemberAnswersToQuestionId(updatedCurTeam, curQuestion.id)
+        setCorrectAnswer(ModelHelper.getCorrectAnswer(currentQuestion))
+        const teamAnswers = ModelHelper.getBasicTeamMemberAnswersToQuestionId(updatedCurTeam, currentQuestion.id)
+
+        const correctAnswer = ModelHelper.getCorrectAnswer(currentQuestion)
+        let originalScore = gameSession?.teams?.find(teamElement => teamElement.id === team.id).score 
 
-        // seeing this useEffect run in screens further up the navigation stack so I'm putting this if statement in temporarily so we don't keep doing api calls
-        if (gameSession.currentState === GameSessionState.PHASE_1_RESULTS || gameSession.currentState === GameSessionState.PHASE_2_RESULTS)
-          totalScore = calculateTotalScore(gameSession, curQuestion, updatedCurTeam)
+        setTotalScore(calculateTotalScore(gameSession, currentQuestion, updatedCurTeam))
         
-          if (!isNullOrUndefined(teamAnswers) && teamAnswers.length > 0) {
-            // User has answered both phases
-            if (teamAnswers.length > 1) {
-                const answer = teamAnswers[1]
-                setAnswer(answer)
-            }
-            setAnswer(teamAnswers[0])
+        if (!isNullOrUndefined(teamAnswers) && teamAnswers.length > 0) {
+            if (phaseNo === 1)
+              setAnswer(teamAnswers.reduce(teamAnswer => teamAnswer.isChosen === true))
+            else 
+              setAnswer(teamAnswers.reduce(teamAnswer => teamAnswer.isTrickAnswer === true))
         } else {
             setSelectedAnswer({
                 id: Number.MIN_VALUE,
                 text: Number.MIN_VALUE,
             })
         }
-        setSelectedTrickAnswer(ModelHelper.getSelectedTrickAnswer(updatedCurTeam, curQuestion.id))
-        setCorrectAnswer(ModelHelper.getCorrectAnswer(curQuestion))
-        setCurrentQuestion(curQuestion)
+
+        setSelectedTrickAnswer(ModelHelper.getSelectedTrickAnswer(updatedCurTeam, currentQuestion.id))
         setLoadedData(true)
         setTeamInfo(updatedCurTeam, updatedCurTeam.teamMembers[0])
-    }, [gameSession])
+      }, [gameSession?.currentState])
+    )
```

Comment:
this is a fairly heavy useEffect so I only want to update this on currentState changes, and only when the screen is focused. 
 TODO: I think we can continue to optimize this because a lot of the existing calls on the screen are being duplicated in ModelHelper.